File: [23.12]Design-Pattern/Java/src/behavioral/iterator/HamburgerCollection.java
package behavioral.iterator;

public class HamburgerCollection implements Collection{
    Hamburger[] arr;
    private int index;

    public HamburgerCollection(int size) {
        this.arr = new Hamburger[size];
    }

    public void add(Hamburger hamburger){
        if(index<arr.length){
            arr[index++] = hamburger;
        }
    }

    @Override
    public Iterator iterator() {
        return new HamburgerIterator(this.arr);
    }

}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/behavioral/iterator/HamburgerCollection.kt
package behavioral.iterator

class HamburgerCollection(private val size: Int) : Collection {
    private val arr = arrayOfNulls<Hamburger>(size)
    private var index = 0

    fun add(hamburger: Hamburger) {
        if (index < arr.size) {
            arr[index++] = hamburger
        }
    }

    override fun iterator(): Iterator {
        return HamburgerIterator(arr)
    }
}